fileuploader-kiwiirc-plugin/components/SidebarFileList.vue

```diff
@@ -0,0 +1,140 @@
+            <span class="kiwi-filebuffer-status-message">Files recently uploaded to {{currentBuffer}}</span><br><br>
+            <table class="kiwi-filebuffer-table">
+                <tr><th>Nick</th><th>Link</th><th>Time</th></tr>
+                <tr v-for="(file, idx) in fileList" :key="file">
+                    <td class="kiwi-filebuffer-table-td">
+                        {{file.nick}}
+                    </td>
+                    <td class="kiwi-filebuffer-table-td">
+                        <a
+                            :href="file.url"
+                            @click.prevent.stop="loadContent(file.url)"
+                            class="kiwi-filebuffer-anchor"
+                        >
+                            {{getFileName(file.url)}}
+                        </a>
+                    </td>
+                    <td class="kiwi-filebuffer-table-td kiwi-filebuffer-time">
+                        {{file.time}}
+                    </td>
+                </tr>
+            </table>
+        </div>
+    </div>
+</template>
+
+<script>
+
+'kiwi public';
+
+export default {
+    data() {
+        return {
+            settings: kiwi.state.setting('fileuploader'),
+        };
+    },
+    methods: {
+        getFileName(file) {
+            file = decodeURI(file);
+            let name = file.split('/')[file.split('/').length-1];
+            if (name.length >= 25) {
+                name = name.substring(0, 18) + '...' + name.substring(name.length - 4);
+            }
+            return name;
+        },
+        loadContent(url) {
+            kiwi.emit('mediaviewer.show', url);
+        },
+        messageHandler() {
+            setTimeout(() => this.sharedFiles(kiwi.state.getActiveBuffer()), 1100)
+        },
+        sharedFiles(buffer) {
+            let returnArr = []
+            let messages = buffer.getMessages()
+            let tmp = buffer.message_count
+            for(let i = 0; i < messages.length; i++) {
+                let e = messages[i]
+                if (e.message.indexOf(this.settings.server) !== -1) {
+                    let time = new Intl.DateTimeFormat('default', { hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(new Date(e.time))
+                    let link = {
+                        url: e.message.substring(e.message.indexOf(this.settings.server)),
+                        nick: e.nick,
+                        time
+                    };
+                    link.url = link.url.split(' ')[0].split(')')[0]
+                    returnArr.push(link);
+                }
+            }
+
+            // comment out the following line to include duplicates
+            returnArr = _.uniqBy(returnArr, 'url')
+
+            kiwi.emit('files.listshared', { fileList: returnArr, buffer })
+            return returnArr
+        },
+    },
+    computed: {
+        currentBuffer() {
+            this.sharedFiles(kiwi.state.getActiveBuffer())
```

Comment:
Should this sharedFiles() call be here anymore?

Reply:
Nope. baleeted.